### PR TITLE
Migrate tests/samples/tools to portable

### PR DESCRIPTION
**Pull request by pakrym**
@davidfowl @anurse @pranavkm

### CODE REVIEW

File: samples/ActionConstraintSample.Web/project.json [hunk 1]
```diff
@@ -6,18 +6,20 @@
   "dependencies": {
     "Microsoft.AspNetCore.Server.IISIntegration": "1.0.0-*",
     "Microsoft.AspNetCore.Mvc": "1.0.0-*",
-    "Microsoft.AspNetCore.Server.Kestrel": "1.0.0-*",
-    "Microsoft.NETCore.Platforms": "1.0.1-*"
```
Comment (davidfowl) [inline samples/ActionConstraintSample.Web/project.json]: This is still required

**Comment by ryanbrandenburg** [comment]
Looks good to me other than @davidfowl 's complaint. I say :shipit:

File: samples/ActionConstraintSample.Web/project.json (hunk 1 above)
Comment (pakrym) [inline samples/ActionConstraintSample.Web/project.json]: For net451?

**Comment by ryanbrandenburg** [comment]
@pakrym this appears to be a (I assume) more correct version of some of the stuff I was doing in #4429, once you get that thing worked out with @davidfowl feel free to merge, I'll resolve my conflicts and follow with the rest of the cleanups.

**Comment by ryanbrandenburg** [comment]
Should merge back into `dev` as well, right?

**Comment by javiercn** [comment]
@pakrym Is already working on it

**Pull request merged**